compiler/rustc_resolve/src/late.rs

```diff
@@ -4680,11 +4689,19 @@ impl<'a, 'ast, 'ra, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
                 && let ItemKind::MacroDef(..) = item.kind
             {
                 num_macro_definition_ribs += 1;
+                let res = self.r.local_def_id(item.id).to_def_id();
+                self.ribs[ValueNS].push(Rib::new(RibKind::LookAheadMacroDefinition(res)));
```

Comment:
I'm pretty sure macros 2.0 (`macro m() {}`) should behave like `LookAheadMacroDefinition`s with regards to local variables.
If they are not, then they are buggy.

This test is currently failing.
```rust
#![feature(decl_macro)]

fn main(){
    fn f() -> i32 { 27 }
    let f = || -> i32 { 42 };
    let a = m!();
    macro m() { f() }
    let b = m!();
    assert_eq!(a, b);
}
```